http/headers.ts

```diff
@@ -0,0 +1,153 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+
+// From node-fetch
+// Copyright (c) 2016 David Frank. MIT License.
+const invalidTokenRegex = /[^\^_`a-zA-Z\-0-9!#$%&'*+.|~]/;
+const invalidHeaderCharRegex = /[^\t\x20-\x7e\x80-\xff]/;
+
+interface HeadersInit {
+  [key:string]: string[];
+}
+
+const headerMap = Symbol("header map");
+
+// ref: https://fetch.spec.whatwg.org/#dom-headers
+export class HttpHeaders {
+  private [headerMap]: Map<string, string[]>;
+
+  private _normalizeParams(name: string, value: string): string[] {
+    return [this._normalizeName(name), this._normalizeValue(value)];
+  }
+
+  private _normalizeName(name: string): string {
+    return String(name).toLowerCase();
+  }
+
+  private _normalizeValue(value: string): string {
+    return String(value).trim();
+  }
+
+  // The following name/value validations are copied from
+  // https://github.com/bitinn/node-fetch/blob/master/src/headers.js
+  // Copyright (c) 2016 David Frank. MIT License.
+  private _validateName(name: string): void {
+    if (invalidTokenRegex.test(name) || name === "") {
+      throw new TypeError(`${name} is not a legal HTTP header name`);
+    }
+  }
+
+  private _validateValue(value: string): void {
+    if (invalidHeaderCharRegex.test(value)) {
+      throw new TypeError(`${value} is not a legal HTTP header value`);
+    }
+  }
+
+  constructor(init?: HeadersInit) {
+    this[headerMap] = new Map();
+    if (init === null) {
```

Comment:
The null check is present in the `Headers` [constructor](https://github.com/denoland/deno/blob/master/js/headers.ts).  I agree with your point, and left it alone because the null check and optional `init` were present during https://github.com/denoland/deno/commit/c0492ef061afd5af2044d5952432d223615841a7#diff-cee0e68f0ac7840274b5171cf4df9dbe, which I believe was the constructor's origin.